src/main/java/io/jenkins/plugins/ml/utils/Dumper.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2020 Loghi Perinpanayagam.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package io.jenkins.plugins.ml.utils;
+
+import hudson.FilePath;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Dumper- A helping tool for save html or image files in the workspace
+ */
+public final class Dumper {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Dumper.class);
+    private static final DateTimeFormatter FORMAT_OBJ = DateTimeFormatter.ofPattern("dd-MM-yyyy-hh-mm-ss");
+
+    private Dumper() {
+    }
+
+    /**
+     * Dump html output from Zeppelin API as html files under the @param foldername .
+     *
+     * @param data       the data
+     * @param foldername the folder name
+     * @param ws         the workspace
+     * @throws IOException          the io exception
+     * @throws InterruptedException the interrupted exception
+     */
+    public static void dumpHtml(String data, String foldername, FilePath ws)
+            throws IOException, InterruptedException {
+        LocalDateTime dateObj = LocalDateTime.now();
+        String filename = File.separator + FORMAT_OBJ.format(dateObj) + ".html";
+        FilePath dumpPath = new FilePath(ws, foldername + filename);
+        dumpPath.write(data, "UTF-8");
+        // enable javascript to run
+        System.setProperty("hudson.model.DirectoryBrowserSupport.CSP", "");
```

Comment:
We cannot change a system-wide property in the plug-in. Once we `setProperty` here, I assume it's changed for the rest of the Jenkins process execution.

A system administrator could get confused (and very angry) after realizing one of the plug-ins is altering the environment variable (especially if they have automation in place to set it to some other value). There appears to be some other possible ways of working around the issue we had with JavaScript (from reading [this doc](https://wiki.jenkins.io/display/JENKINS/Configuring+Content+Security+Policy)).

Maybe worth exploring other options, to avoid security issues in the plug-in (which remove it from update center when reported)